clang/test/CIR/CodeGen/builtin-arm-ldrex.c

```diff
@@ -0,0 +1,21 @@
+// RUN: %clang_cc1 -triple aarch64-none-linux-android24  -fclangir -emit-cir -mmlir --mlir-print-ir-before=cir-lowering-prepare %s -o %t.cir 2>&1 | FileCheck %s
```

Comment:
For the test name, let's go similar to the OG one here `clang/test/CodeGen/builtins-arm-exclusive.c`. When implementing those, I suggest you take a look at the existing OG tests and reuse them if possible.

Note how they test the variations for different int sizes, we do the same in `clang/test/CIR/CodeGen/aarch64-neon-intrinsics.c`, please add more tests for different int types.